File: 0092. Reverse Linked List II/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;

        ListNode bl = dummy;
        ListNode l = head;

        for(int i=1;i<left;i++){
            bl = bl.next;
            l = l.next;
        }

        ListNode prev = null;

        for(int i=0;i<(right-left)+1;i++){
            ListNode next = l.next;

            l.next = prev;
            prev = l;
            l = next;
        }

        bl.next.next = l;
        bl.next = prev;

        return dummy.next;

    }
}

File: 0092. Reverse Linked List II/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseBetween(self, head, left, right):
        """
        :type head: Optional[ListNode]
        :type left: int
        :type right: int
        :rtype: Optional[ListNode]
        """
        if head is None:
            return None

        dummy = ListNode(0)
        dummy.next = head

        l = head
        bl = dummy

        for i in range(left-1):
            bl = l
            l = l.next

        prev = None

        for i in range((right-left)+1):
            next = l.next

            l.next = prev
            prev = l
            l = next

        bl.next.next = l
        bl.next = prev

        return dummy.next

        